src/content/articles/kubernetes-for-scala-developers-part-one/index.mdx

```diff
@@ -0,0 +1,1052 @@
+  secret-two: bW9uZ29wYXNzd29yZA==
+
+# ConfigMap
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: workload-config
+data:
+  config-one: "config-value"
+
+# Deployment
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: workload-api
+  labels:
+    demo: k8s-auto-scaling
+spec:
+  selector:
+    matchLabels:
+      type: api
+      language: scala
+      framework: play-framework
+      service: workload
+  minReadySeconds: 1
+  progressDeadlineSeconds: 1200
+  revisionHistoryLimit: 5
+  strategy: # Deployment strategy
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+  template: # Pod definition
+    metadata:
+      labels:
+        type: api
+        language: scala
+        framework: play-framework
+        service: workload
+    spec:
+      containers:
+        - name: workload-api
+          image: aidocking/pfpw:20.20.31
+          ports:
+            - containerPort: 9000
+          env:
+            - name: CONFIG_ONE
+              valueFrom:
+                configMapKeyRef:
+                  name: workload-config
+                  key: config-one
+            - name: SECRET_ONE
+              valueFrom:
+                secretKeyRef:
+                  name: workload-secret
+                  key: secret-one
+
+          livenessProbe:
+            httpGet:
+              path: /live
+              port: 9000
+            failureThreshold: 3
+            timeoutSeconds: 2
+            periodSeconds: 10
+            initialDelaySeconds: 15
+
+          startupProbe:
+            httpGet:
+              path: /startup
+              port: 9000
+            failureThreshold: 30
+            periodSeconds: 15
+            timeoutSeconds: 2
+            initialDelaySeconds: 45
+
+          readinessProbe:
+            httpGet:
+              path: /ready
+              port: 9000
+            failureThreshold: 10
+            periodSeconds: 15
+            timeoutSeconds: 10
+
+          resources:
+            limits:
+              memory: "2.5Gi"
+              cpu: "2"
+
+# api service
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: workload-service
+  labels:
+    service: workload
+spec:
+  type: ClusterIP
+  selector:
+    type: api
+    language: scala
+    framework: play-framework
+    service: workload
+  ports:
+    - port: 9002
+      protocol: TCP
+      targetPort: 9000
+
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: api-workload-ingress
+  annotations:
+    'spec.ingressClassName': "nginx"
+    ingress.kubernetes.io/ssl-redirect: "false"
+    nginx.ingress.kubernetes.io/ssl-redirect: "false"
+spec:
+  rules:
+    - host: k8splayworkload.com
+      http:
+        paths:
+          - path: /
+            pathType: ImplementationSpecific
+            backend:
+              service:
+                name: workload-service
+                port:
+                  number: 9002
+
+---
+# workload auto-scaling strategy
```

Comment:
don't add the autoscaling until you start talking about it

Reply:
I have put it there because I explains it just below: <<#### Auto-scaling
We have added a horizontal auto-scaling policy for this deployment.>>